specs/UserActivityHistory/UserActivityHistoryAPI.md

```diff
@@ -0,0 +1,243 @@
+
+## UserActivityHistory.GetAppsWithUserActivityAsync method
+
+This method is an asynchronous version of the `GetAppsWithUserActivity` method.
+
+## UserActivityHistoryItem class
+
+This class represents a single item in the user's activity history. It contains properties that
+describe in what app the activity occurred, what the nature of the activity was, the URI of the
+resource involved in the activity (e.g., a document, a webpage, a video, etc.), the URI that
+can be used to bring back the state the user left the activity in, and the times when the user
+started the activity and ended the activity.
+
+If the user performed the same activity multiple times, there will be multiple
+`UserActivityHistoryItem` objects returned, each with different start and end times.
+
+## UserActivityHistoryItem.AppName property
+
+This property contains the name of the app in which the activity occurred.
+
+## UserActivityHistoryItem.ActivityId property
+
+This property contains the ID of the activity, which can be used to collate multiple sessions
+of the same activity. For example, if the user watched a movie in multiple sessions, the
+`ActivityId` property can be used to identify how long in total the user spent watching that movie.
+
+## UserActivityHistoryItem.DisplayText property
+
+This property contains a string that is how the app chose to describe the activity. For example,
+if the activity was reading the contents of a webpage, this property might contain the webpage's title.
+
+## UserActivityHistoryItem.ContentType property
+
+This property contains the MIME type of the content being interacted with. For example, if the user
+was looking at a PNG image, this property would contain the string "image/png".
+
+## UserActivityHistoryItem.ContentUri property
+
+This property contains the URI of the content being interacted with. For example, if the user was
+looking at a webpage, this property would contain the URI of that webpage.
+
+## UserActivityHistoryItem.ActivationUri property
+
+This property contains the URI that can be used to bring back the state the user left the activity in.
+For example, if the user was looking at a webpage, this property would contain the URI of that webpage
+with additional information such as what page the user was on, what their scroll position was, etc.
+
+## UserActivityHistoryItem.StartTime property
+
+This property contains the time when the user started the activity session.
+
+## UserActivityHistoryItem.EndTime property
+
+This property contains the time when the user ended the activity session.
+
+## UserActivityHistoryQuery class
+
+This class is used to specify criteria for what portion of the user's activity history you want to
+retrieve. It allows you to specify keywords to search for, content types to filter by, and time ranges
+to filter by.
+
+## UserActivityHistoryQuery.Keywords property
+
+This property is an array of keywords, each of which is used to lexically search against the
```

Comment:
there is no semantic search? If there's a misspelling or alternate spelling of a keyword, the system won't find it?
e.g. en-US "vacation", en-UK "holiday"

Reply:
Correct - we're expecting that the app will have done the heavy lifting of semantic parsing, etc.  The specific scenario we're picturing for this API is an agentic AI app that will construct the parameters to pass into this API.

We certainly might be able to add semantic search later, but that is currently not in scope for this API.

Reply:
thanks! that's fine for a v1, but tagging @andrewleader to consider this as an improvement